docs/xks/developer-guide/observability.md

```diff
@@ -0,0 +1,244 @@
+This means that `kubectl logs` and Datadog will display the same information.
+
+Check the official [Datadog Logging Documentation](https://docs.datadoghq.com/agent/kubernetes/log/?tab=daemonset) for more detailed information.
+
+### Metrics
+
+Datadog can collect exposed Prometheus or OpenMetrics from your application. In simple terms this means that the application needs to expose an
+endpoint which the Datadog agent can scrape to get the metrics. All that is required is that the Pod contains annotations which tells Datadog where to
+find the metrics.
+
+Given that your application is exposing metrics on port 8080 your pod should contain the following annotations.
+
+```yaml
+annotations:
+    ad.datadoghq.com/prometheus-example.instances: |
+      [
+        {
+          "prometheus_url": "http://%%host%%:8080/metrics"
+        }
+      ]
+```
+
+Check the official [Datadog Metrics Documentation](https://docs.datadoghq.com/agent/kubernetes/prometheus/) for more detailed information.
+
+### Tracing
+
+Datadog tracing is done with Application Performance Monitoring (APM), which sends traces from an application to Datadog. For traces to work the
+application needs to be configured with the language specific libraries. Check the [Language
+Documentation](https://docs.datadoghq.com/tracing/setup_overview/) for language specific instructions. Some of the languages that are supported are.
+
+- Golang
+- C#
+- Java
+- Python
+
+Configure your Deployment with the `DD_AGENT_HOST` environment for the APM agent to know where to send the traces.
+
+```.yaml
+apiVersion: apps/v1
+kind: Deployment
+spec:
+  containers:
+    - env:
+        - name: DD_AGENT_HOST
+          valueFrom:
+            fieldRef:
+              fieldPath: status.hostIP
+```
+
+Check the official [Datadog Tracing Documentation](https://docs.datadoghq.com/agent/kubernetes/apm/?tab=helm) for more detailed information.
+
+### networkpolicy datadog
+
+When using XKF and your cluster have enabled datadog the tenant namespace will get a networkpolicy automatically that allows egress for tracing.
+
+You can view these rules by typing
+
+```shell
+kubectl get networkpolicies
+```
+
+## Opentelemetry
+
+To gather opentelemtry data we rely on the [grafana agent operator](https://github.com/grafana/agent).
+
+### Metrics
+
+To gather metrics data we use servicemonitors or podmonitors that is managed in XKF using the [prometheus-operator](https://github.com/prometheus-operator/prometheus-operator/).
+
+The prometheus-operator have great [getting started guide](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md)
+But if you want a quick example you can look below.
+
+The most important part for you as a developer you have to label the pod/service monitor yaml with `xkf.xenit.io/monitoring: tenant`,
+else the grafana agent won't find the rule to gather the metric.
```

Comment:
`else the grafana agent won't find the rule to gather the metric.` should I explain even more why?

Reply:
Either " The most important part for you as a developer you have to label the pod/service monitor yaml with `xkf.xenit.io/monitoring: tenant`" means that you should insert exactly that label, in which case it is not so important, because you are anyway just copying the example. Or it means you should replace `tenant` with your namespace name, in which case the text should say that.

However, in the example below, you will have to update this section:
```
  selector:
    matchLabels:
      app.kubernetes.io/name: app1
      app.kubernetes.io/instance: app1
```
and I think it might be good to explain what `app1` should be changed to, probably also the distinction between `name` and `instance`?

Reply:
I will try to change the wording. 
I don't think that we should try to rewrite the kubernetes docs, if we do that it's a slippery slope and it will never finish. I assume that developers knows this. But if they don't I will add a link to the official kubernetes docs that explains it.